src/tribler/core/components/watch_folder/tests/test_watch_folder.py

```diff
@@ -120,3 +120,34 @@ async def test_watch_folder_no_crash_exception(watch_folder: WatchFolder):
     # Test that errors raised in `_check_watch_folder` reraise as `NoCrashException`
     with pytest.raises(NoCrashException):
         await watch_folder._check_watch_folder_handle_exceptions()
+
+
+async def test_watch_folder_invalid_dir(watch_folder: WatchFolder, tmp_path):
+    """ Test that the watch folder is disabled when the directory is invalid"""
+    watch_folder.settings.put_path_as_relative(
+        property_name='directory',
+        value=Path('incorrect path'),
```

Comment:
**nitpick:** This `'incorrect path'` is not actually incorrect; it just does not exist. Correspondingly, this path does not trigger `OSError` inside the `is_valid` method. But it is probably OK as handling OSError is covered by `test_is_invalid_by_os_exception`. As I understand it, making a path that is universally syntactically incorrect on all OSes is a bit tricky.

Reply:
You are right :) What if we use this construction:

```python
async def test_watch_folder_invalid_dir(watch_folder: WatchFolder, tmp_path):
    """ Test that the watch folder is disabled when the directory is invalid"""
    watch_folder.settings.put_path_as_relative(
        property_name='directory',
        value=Path('# incorrect path'),
        state_dir=Path(tmp_path)
    )

    assert not await watch_folder._check_watch_folder()

```